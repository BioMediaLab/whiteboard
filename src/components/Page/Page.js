import React from 'react'
import { Page, Layout } from '@shopify/polaris'

export default ({ children, hasLayout = false, ...rest }) => {
  if (!hasLayout) return <Page {...rest}>{children}</Page>

  return (
    <Page {...rest}>
      <Layout>{children}</Layout>
    </Page>
  )
}

export const PagePrimary = ({ children }) => {
  return <Layout.Section>{children}</Layout.Section>
}

export const PageSecondary = ({ children }) => {
  return <Layout.Section secondary>{children}</Layout.Section>
}
