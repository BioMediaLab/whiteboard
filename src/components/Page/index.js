export { default, PagePrimary, PageSecondary } from './Page'
