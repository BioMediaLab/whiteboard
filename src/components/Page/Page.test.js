import React from 'react'
import Page, { PageContent, PageHeader, PageFooter, PageSidebar } from './'

describe('Page', () => {
  test('can render', () => {
    const component = render(
      <Page>
        <p>A little content for the page.</p>
      </Page>
    )

    expect(component).toMatchSnapshot()
  })

  test('can render with all', () => {
    const component = render(
      <Page>
        <PageHeader>
          <h1>Example</h1>
        </PageHeader>
        <PageSidebar>
          <p>Sidebar Content</p>
        </PageSidebar>
        <PageContent>
          <p>A little content for the page.</p>
        </PageContent>
        <PageFooter>
          <p>Footer Content</p>
        </PageFooter>
      </Page>
    )

    expect(component).toMatchSnapshot()
  })

  test('can render with header and content', () => {
    const component = render(
      <Page>
        <PageHeader>
          <h1>Example</h1>
        </PageHeader>
        <PageContent>
          <p>A little content for the page.</p>
        </PageContent>
      </Page>
    )

    expect(component).toMatchSnapshot()
  })
})
