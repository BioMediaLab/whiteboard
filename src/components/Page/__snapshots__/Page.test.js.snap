// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page can render 1`] = `
<div>
  <div
    class="Page"
  >
    <p>
      A little content for the page.
    </p>
  </div>
</div>
`;

exports[`Page can render with all 1`] = `
<div>
  <div
    class="Page"
  >
    <div
      class="PageHeader"
    >
      <h1>
        Example
      </h1>
    </div>
    <div
      class="PageSidebar"
    >
      <p>
        Sidebar Content
      </p>
    </div>
    <div
      class="PageContent"
    >
      <p>
        A little content for the page.
      </p>
    </div>
    <div
      class="PageFooter"
    >
      <p>
        Footer Content
      </p>
    </div>
  </div>
</div>
`;

exports[`Page can render with header and content 1`] = `
<div>
  <div
    class="Page"
  >
    <div
      class="PageHeader"
    >
      <h1>
        Example
      </h1>
    </div>
    <div
      class="PageContent"
    >
      <p>
        A little content for the page.
      </p>
    </div>
  </div>
</div>
`;
